apps/src/code-studio/components/header/ProjectInfo.jsx

```diff
@@ -0,0 +1,37 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {connect} from 'react-redux';
+
+import ProjectHeader from './ProjectHeader';
+import MinimalProjectHeader from './MinimalProjectHeader';
+import ProjectBackedHeader from './ProjectBackedHeader';
+import LevelBuilderSaveButton from './LevelBuilderSaveButton';
+
+import {possibleHeaders} from '../../headerRedux';
+
+const headerComponents = {
+  [possibleHeaders.project]: ProjectHeader,
+  [possibleHeaders.minimalProject]: MinimalProjectHeader,
+  [possibleHeaders.projectBacked]: ProjectBackedHeader,
+  [possibleHeaders.levelBuilderSave]: LevelBuilderSaveButton
+};
```

Comment:
I really enjoyed this elegant way to make the inner components mutually exclusive.

Reply:
+1! Agree that life would be worse if you had to use inline control flow exclusively here.

Reply:
Thanks! I was actually halfway through implementing an inline control flow version when I realized this would be _way_ easier